src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs

```diff
@@ -275,6 +276,23 @@ public string EndWriteTagHelperAttribute()
             return content;
         }
 
+        public virtual void PushWriter(TextWriter writer)
```

Comment:
Agree, just thinking to default to non-visible first and upon customer feedback change it if we need to.